src/modules/acbs/acbs-bundle-information.service.create-bundle-information.test.ts

```diff
@@ -357,5 +358,104 @@ describe('AcbsBundleInformationService', () => {
         await expect(createBundleInformationPromise).rejects.toHaveProperty('errorBody', JSON.stringify(errorBody));
       });
     });
+
+    describe('creating a FacilityFeeAmountTransaction', () => {
+      const facilityIdentifier = valueGenerator.facilityId();
+
+      const facilityFeeAmountTransactionMessage: FacilityFeeAmountTransaction = {
+        $type: 'FacilityFeeAmountTransaction',
+        AccountOwnerIdentifier: valueGenerator.string(),
+        EffectiveDate: valueGenerator.dateTimeString(),
+        FacilityIdentifier: facilityIdentifier,
+        FacilityFeeTransactionType: {
+          TypeCode: valueGenerator.nonnegativeInteger(),
+        },
+        IsDraftIndicator: valueGenerator.boolean(),
+        LenderType: {
+          LenderTypeCode: valueGenerator.enumValue(ENUMS.LENDER_TYPE_CODES),
+        },
+        LimitKeyValue: valueGenerator.acbsPartyId(),
+        LimitType: {
+          LimitTypeCode: valueGenerator.string(),
+        },
+        SectionIdentifier: valueGenerator.string(),
+        SegmentIdentifier: valueGenerator.string(),
+        TransactionAmount: valueGenerator.nonnegativeFloat(),
+      };
+      const acbsRequestBodyToCreateFacilityFeeAmountTransaction: AcbsCreateBundleInformationRequestDto<FacilityFeeAmountTransaction> = {
+        ...acbsRequestBodyToCreateFacilityActivationTransaction,
+        BundleMessageList: [facilityFeeAmountTransactionMessage],
+      };
+      const expectedHttpServicePostArgsForFacilityFeeAmountTransaction = expectedHttpServicePostArgsWithBody(
+        acbsRequestBodyToCreateFacilityFeeAmountTransaction,
+      );
+
+      it('throws an AcbsResourceNotFoundException if ACBS responds with a 400 error that is a string containing "Facility does not exist"', async () => {
```

Comment:
Should we create an assertion where `SegmentIdentifier` is invalid?

Reply:
For client this field is known as `period`. We validate all client input in api-tests using helper.

    withStringFieldValidationApiTests({
      fieldName: 'period',
      length: 2,
      generateFieldValueOfLength: (length: number) => valueGenerator.string({ length }),
      validRequestBody: increaseAmountRequest,
      makeRequest,
      givenAnyRequestBodyWouldSucceed,
    });